pkg/registry/deployment/strategy.go

```diff
@@ -61,6 +61,9 @@ func (deploymentStrategy) AllowCreateOnUpdate() bool {
 
 // PrepareForUpdate clears fields that are not allowed to be set by end users on update.
 func (deploymentStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newDeployment := obj.(*extensions.Deployment)
```

Comment:
I see that we do the same thing in PrepareForCreate for other resources.
Ex: pkg/registry/ingress/strategy.go